Creates TimeRemaing.ios.tsx and updates TimeRemaining API (#564)
artsy/palette: README.md

When adding a new component to Palette, it's important to be aware that this library is used on the web as well as in React Native, via Emission, and therefore must follow a few rules in terms of structure, namely:

> > If a mobile component has platform-specific features, that code has to live in a `Component.ios.tsx` file. Or if a web component has browser-only features, then a `Component.ios.tsx` file must be created, or React Native will error out. If the code between a web and native component is identical, a `.ios.tx` file isn’t needed. If some code can be shared between platforms, then that shared code should live in a `Component.shared.tsx` file.

Reviewer: Some editing suggestions to give this paragraph a little more breathing room:

```suggestion
- If a mobile component has platform-specific features, that code must live in a `Component.ios.tsx` file.
- If a web component has browser-only features, then an empty `Component.ios.tsx` file must be created. Without this file, React Native will error out.
- If the code between a web and native component is identical, a `.ios.tx` file is not needed.
- If some code can be shared between platforms, then that shared code should live in a `Component.shared.tsx` file.
```

In the previous paragraph we say "must follow a few rules in terms of structure", so I think it makes sense to turn this into a list of rules instead of one paragraph.

> > If a mobile component has platform-specific features, that code has to live in a `Component.ios.tsx` file. Or if a web component has browser-only features, then a `Component.ios.tsx` file must be created, or React Native will error out. If the code between a web and native component is identical, a `.ios.tx` file isn’t needed. If some code can be shared between platforms, then that shared code should live in a `Component.shared.tsx` file.

Reviewer 2: 👍
